pywr/parameters/_parameters.pyx

```diff
@@ -1849,7 +1849,13 @@ def load_parameter(model, data, parameter_name=None):
         parameter = data
     else:
         # parameter is dynamic
-        parameter_type = data['type']
+
+        try:
+            parameter_type = data['type']
+        except KeyError:
+            data_summary = str(data)[0:100]
+            raise KeyError(f"Unable to find key 'type' in {data_summary}")
```

Comment:
Why don't we just pass the whole of `data` as a second argument?
```python
raise KeyError('Parameter data contains no "type" key.', data)
```

Reply:
You mean into the exception? Data can be very big. I had data in there, but the screen just filled up with a big data frame. It kind of defeated the purpose of the re-raise

Reply:
Oh I forgot about the in-line data. I was worried that the first 100 characters might not display what you want (e.g. if the keys were in the wrong order). We *could* have a custom exception class that knew how to interpret `data` and produced a sensible message passed on its contents. What do you think?

Reply:
> Oh I forgot about the in-line data. I was worried that the first 100 characters might not display what you want (e.g. if the keys were in the wrong order). We _could_ have a custom exception class that knew how to interpret `data` and produced a sensible message passed on its contents. What do you think?

This makes sense to me, as the 1st 100 characters is somewhat arbitrary and may not tell you anything. Want me to add that to this MR or create a separate issue?

Reply:
Create a separate issue if not solving now. What should we do with this now?